src/main/java/newbank/server/NewBank.java

```diff
@@ -42,10 +45,24 @@ private void addTestData() {
    * @param username The customer's username
    * @param password The customer's password
    */
-  public void addCustomer(final String username, final String password) {
+  public void addCustomer(final String username, final String password)
+      throws DuplicateCustomerException, PasswordInvalidException {
+    if (customers.containsKey(username)) {
+      throw new DuplicateCustomerException();
```

Comment:
Perfect!

Reply:
Perhaps, to be consistent, it should be `CustomerDuplicateException`. Again - I do realize it doesn't sounds most natural.

Reply:
It does not read well I don't think.